spec/controllers/concerns/hyrax/works_controller_behavior_spec.rb

```diff
@@ -1,18 +1,29 @@
 # frozen_string_literal: true
 
 RSpec.describe Hyrax::WorksControllerBehavior, :clean_repo, type: :controller do
-  let(:paths) { Rails.application.routes.url_helpers }
+  let(:paths) { controller.main_app }
   let(:title) { ['Comet in Moominland'] }
   let(:work)  { FactoryBot.valkyrie_create(:hyrax_work, alternate_ids: [id], title: title) }
   let(:id)    { '123' }
 
-  before(:context) { Hyrax.config.register_curation_concern(Hyrax::Test::SimpleWork) }
+  let(:main_app_routes) do
+    ActionDispatch::Routing::RouteSet.new.tap do |r|
+      r.draw do # draw minimal routes for this controller mixin
+        mount Hyrax::Engine, at: '/'
+        namespaced_resources 'hyrax/test/simple_work_legacy', except: [:index]
+        devise_for :users
+        resources :solr_documents, only: [:show], path: '/catalog', controller: 'catalog'
+      end
+    end
+  end
 
-  after(:context) do
-    config = Hyrax.config
-    types  = config.registered_curation_concern_types - ["Hyrax::Test::SimpleWork"]
+  before do
```

Comment:
Is this working now? I'd imagine extracting a general support module might be helpful.

Reply:
i think it's working! nothing else is yet using similar behavior, but i'm :+1: on extracting before we end up duplicating this elsewhere.